src/libretro/FrontendBridge.cpp

```diff
@@ -345,3 +347,178 @@ void CFrontendBridge::LocationDeinitialized(void)
 {
   // Not implemented
 }
+
+const char *CFrontendBridge::GetPath(retro_vfs_file_handle *stream)
+{
+  if (stream == nullptr)
+    return "";
+
+  FileHandle *fileHandle = reinterpret_cast<FileHandle*>(stream);
+
+  return fileHandle->path.c_str();
+}
+
+retro_vfs_file_handle *CFrontendBridge::OpenFile(const char *path, unsigned mode, unsigned hints)
+{
+  if (path == nullptr)
+    return nullptr;
+
+  std::unique_ptr<FileHandle> fileHandle(new FileHandle{ path });
+  fileHandle->file.reset(new kodi::vfs::CFile);
+
+  bool bRead = true;
+
+  switch (mode)
+  {
+  case RETRO_VFS_FILE_ACCESS_READ:
+    bRead = true;
+    break;
+  case RETRO_VFS_FILE_ACCESS_WRITE:
+    bRead = false;
+    break;
+  case RETRO_VFS_FILE_ACCESS_READ_WRITE:
+    return nullptr; // TODO
+  case RETRO_VFS_FILE_ACCESS_UPDATE_EXISTING:
+    return nullptr; // TODO
```

Comment:
Looks good, only with the two I'm not sure myself whether this is already possible or requires an API update. Think update would make sense to make its use easier to understand.

Reply:
Was this functionality added in your VFS work?

Reply:
Maybe it works already, looked at https://github.com/xbmc/xbmc/blob/master/xbmc/platform/posix/filesystem/PosixFile.cpp#L74 and seems with `O_RDWR` for `RETRO_VFS_FILE_ACCESS_READ_WRITE` already supported, also in Windows seems it set.

About `RETRO_VFS_FILE_ACCESS_UPDATE_EXISTING` maybe also works if on second function value the `overwrite` is set to true, only maybe good to compare there between Kodi and Original about inited file positions.
The https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-dev-kit/include/kodi/Filesystem.h#L1877.

Reply:
We can open for read and open for write, but can we open for read-write?

Reply:
Yes, the `OpenFileForWrite` should then also have always read access too.

Reply:
OK, I've updated the logic for the appropriate flags